ambari-common/src/main/python/resource_management/libraries/execution_command/execution_command.py

```diff
@@ -251,6 +390,18 @@ def need_refresh_topology(self):
   def check_only_update_files(self):
     return self.__get_value('commandParams/update_files_only', False)
 
+  def get_ha_node_role(self):
```

Comment:
get_namenode_ha_node() or get_desired_namenode_role() instead? There are other services that support HA too and see they might assume that they can use this function for it.

Reply:
but my concern is the namenode is used, is this specific to hdfs?, suppose we should make service agnostic

Reply:
yes but this property is service specific and hence we have to treat this as an exception :)

Reply:
ok, so I will update the code to get_desired_namenode_role()?